File: BillsPCBackend/src/main/java/com/revature/service/RegisterService.java
package com.revature.service;

import com.revature.dao.TrainerDaoImp;
import com.revature.dao.RegisterDAO;

public class RegisterService
{
	private static TrainerDaoImp dao = new TrainerDaoImp();
	
    public static boolean register(String username, String password, String email, String role)
    {
    	if(dao.retrieveTrainerByName(username)==null) {
    		RegisterDAO.register(username, password, email, role);
    		return true;
    	}
    	
        
        return false;
    }
}

File: BillsPCFrontEnd/bills-pc-app/src/app/services/register.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs/Observable';

@Injectable()
export class RegisterService {

  constructor(private http: HttpClient) { }

  register(username: string, password: string, email: string, role: string): Observable<boolean> {
    const body = new HttpParams().set('username', username).set('password', password).set('email', email).set('role', role);
    return this.http.post<boolean>('register', body);
  }
}
